core/tracing/tracer.go

```diff
@@ -0,0 +1,78 @@
+package tracing
+
+import (
+	"time"
+
+	"go.uber.org/fx/core/ulog"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber-go/tally"
+	jaegerconfig "github.com/uber/jaeger-client-go/config"
+)
+
+// InitGlobalTracer instantiates a new global tracer
+func InitGlobalTracer(
+	cfg *jaegerconfig.Configuration,
+	serviceName string,
+	logger ulog.Log,
+	scope tally.Scope,
+) (opentracing.Tracer, error) {
+	appCfg := cfg
+	var reporter *jaegerReporter
+	if cfg == nil || !cfg.Disabled {
+		appCfg = loadAppConfig(cfg, logger)
+		reporter = &jaegerReporter{
+			reporter: scope.Reporter(),
+		}
+	}
+	tracer, closer, err := appCfg.New(serviceName, reporter)
+	if err != nil {
+		return tracer, err
+	}
+	defer closer.Close()
+	opentracing.InitGlobalTracer(tracer)
+	return tracer, nil
+}
+
+func loadAppConfig(cfg *jaegerconfig.Configuration, logger ulog.Log) *jaegerconfig.Configuration {
+	appCfg := cfg
+	if appCfg == nil {
+		appCfg = &jaegerconfig.Configuration{}
+	}
+	if appCfg.Logger == nil {
+		jaegerlogger := &jaegerLogger{
+			log: logger,
+		}
+		appCfg.Logger = jaegerlogger
+	}
+	return appCfg
+}
+
+type jaegerLogger struct {
+	log ulog.Log
```

Comment:
The jaeger logger has a different interface, this is a conversion layer from jaeger logger to uberfx logger interface.